packages/components/ovh-shell/src/plugin/routing/orchestrator.ts

```diff
@@ -0,0 +1,164 @@
+    iframe: ILocation,
+    container: ILocation,
+  ) {
+    this.iframe = iframe;
+    this.container = container;
+    this.config = config;
+  }
+
+  static create(
+    config: RoutingConfiguration,
+    iframeWindow: Window,
+    containerWindow: Window,
+  ): Orchestrator {
+    return new Orchestrator(
+      config,
+      {
+        setURL: (href) => iframeWindow.location.replace(href),
+        getURL: () => new URL(iframeWindow.location.href),
+      },
+      {
+        setURL: (href) => containerWindow.location.replace(href),
+        getURL: () => new URL(containerWindow.location.href),
+      },
+    );
+  }
+
+  /**
+   * @returns {URL} The iframe current URL
+   */
+  getIFrameURL() {
+    return this.iframe.getURL();
+  }
+
+  /**
+   * Update the iframe URL to match the container URL.
+   * If the application is standalone, instead redirect to the application.
+   * If the application is unknown, instead redirect to the default application.
+   */
+  updateIframeURL() {
+    const { appId, appHash, appURL } = this.parseContainerURL();
+    if (appURL) {
+      if (appURL.href !== this.getIFrameURL().href) {
+        this.iframe.setURL(appURL.href);
+      }
+    } else {
+      const redirection = this.config.findRedirection(appId);
+      if (redirection) {
+        const target = new URL(redirection.href);
+        target.hash = appHash;
+        this.container.setURL(target.href);
+      } else {
+        this.redirectToContainerHome();
+      }
+    }
+  }
+
+  /**
+   * Update the container URL to match the iframe URL.
+   * If the parsing of iframe URL fails (either invalid URL or the application
+   * is not registered) set the container URL to the default application.
+   */
+  updateContainerURL() {
+    const containerURL = this.container.getURL();
+    const { containerURL: updatedContainerURL } = this.parseIframeURL();
+
+    if (updatedContainerURL) {
+      if (containerURL.href !== updatedContainerURL.href) {
+        this.container.setURL(updatedContainerURL.href);
+      }
+    } else {
+      this.redirectToContainerHome();
+    }
+  }
+
+  /**
+   * Change the container URL to the home page (default application)
+   */
+  redirectToContainerHome() {
+    const defaultAppId = this.config.getDefault()?.id;
+    if (defaultAppId) {
+      const defaultContainerURL = this.container.getURL();
+      defaultContainerURL.hash = `#/${defaultAppId}/`;
+      this.container.setURL(defaultContainerURL.href);
+    }
+  }
+
+  /**
+   * Parse the iframe's URL :
+   * @returns {Object} routing The parsed routing configuration.
+   * @returns {string} routing.id The application's id.
+   * @returns {string} routing.path The application's path.
+   * @returns {string} routing.hash The application's internal routing hash.
+   * @returns {string} routing.appURL The iframe current absolute URL.
+   * @returns {string} routing.containerURL The container URL corresponding to the iframe URL.
+   */
```

Comment:
I think that creating a return type interface for the function might be better than all the jsdoc. Especially that it's reused in other functions in this same file.